### PR TITLE
Changing user collection schema and friends list

### PR DESCRIPTION
**Pull request by ZacMossHK**

### CODE REVIEW

File: controllers/friends.js [hunk 1]
```diff
@@ -5,23 +5,37 @@ const FriendsController = {
...
-          console.log("user added to friends list");
-          res.redirect(303, "/profiles/" + friendUsername);
-        }
+        console.log(err);
+      } else {
+        // update current user's friend list with new friend added
+        newFriendList = currentUser.friends.concat(friendUser._id);
+        User.updateOne(
+          { username: currentUser.username },
+          { $push: { friends: friendUser._id } },
```
Comment (naomischlosser) [inline controllers/friends.js:19]: We might want to change to $addToSet (this only adds a friend to the list if it doesn't yet exist).

File: controllers/friends.js [hunk 2]
```diff
@@ -5,23 +5,37 @@ const FriendsController = {
...
-          console.log(err);
-        } else {
-          req.session.user.friends = newFriendList;
-          console.log("user added to friends list");
-          res.redirect(303, "/profiles/" + friendUsername);
-        }
+        console.log(err);
+      } else {
+        // update current user's friend list with new friend added
+        newFriendList = currentUser.friends.concat(friendUser._id);
```
Comment (naomischlosser) [inline controllers/friends.js:16]: What is this line for if we add something to the database on line 19?

File: controllers/profiles.js [hunk 3]
```diff
@@ -2,19 +2,28 @@ const User = require("../models/user");
...
+          console.log("ProfilePage.index error with User.findOne");
+          console.log(err);
+        } else {
+          const friendsListWithUsernames = user.friends.map(
+            (friend) => friend.username
+          );
+          res.render("profiles/index", {
+            profileUsername: profileUsername,
+            friends: friendsListWithUsernames,
+            fetchUrl: "/friends/requests/new/" + profileUsername,
```
Comment (naomischlosser) [inline controllers/profiles.js:23]: Could you briefly explain where this URL is coming from again?

File: controllers/friends.js (hunk 1 above)
Comment (ZacMossHK) [reply controllers/friends.js:19]: good point, I'll try this and see if it works as well

**Pull request merged**